pymc_marketing/clv/utils.py

```diff
@@ -91,35 +91,40 @@ def _squeeze_dims(x: xarray.DataArray):
     # Monetary value can be passed as a DataArray, with entries per chain and draw or as a simple vector
     if not isinstance(monetary_value, xarray.DataArray):
         monetary_value = to_xarray(customer_id, monetary_value)
-    monetary_value = _squeeze_dims(monetary_value)
 
-    frequency, recency, T = to_xarray(customer_id, frequency, recency, T)
+    monetary_value = _squeeze_dims(monetary_value)
 
     clv = xarray.DataArray(0.0)
 
-    # TODO: Vectorize computation so that we perform a single call to expected_num_purchases
-    prev_expected_num_purchases = _squeeze_dims(
-        transaction_model.expected_num_purchases(
-            customer_id=customer_id,
-            frequency=frequency,
-            recency=recency,
-            T=T,
-            t=0,
+    # TODO: if data attribute added to transaction model,
+    #       transaction_data can be an optional arg for out-of-sample data
+    # TODO: Set these as transaction_model attributes rather than create a dataframe?
+    transaction_data = pd.DataFrame(
```

Comment:
This should probably be using xarray tools not pandas, since the predictive methods work with xarray as well

Reply:
This will need to be included in the IF block mentioned in the comment below, but may require updating the type hints in the transaction models to keep `mypy` happy.

I'll add a TODO for this because I'm not a fan of creating a pandas dataframe here, but don't want to risk a merge conflict with https://github.com/pymc-labs/pymc-marketing/pull/463 either.